src/components/manage/Blocks/Block/EditBlockWrapper.jsx

```diff
@@ -71,6 +87,25 @@ const EditBlockWrapper = (props) => {
               <Icon name={trashSVG} size="18px" />
             </Button>
           )}
+          {showBlockChooser && (
+            <BlockChooserButton
+              data={data}
+              block={block}
+              onInsertBlock={(id, value) => {
+                if (blockHasValue(data)) {
+                  onSelectBlock(onInsertBlock(id, value));
+                } else {
+                  onChangeBlock(id, value);
+                }
+              }}
+              onMutateBlock={onMutateBlock}
```

Comment:
Just realised of this after a closer test. We are mutating the current block if the block is empty. This differs of the current implementation. Maybe that's the reason why the layout editor is "broken"? /cc @avoinea 

This seems quite innocuous change, but it's important.

Reply:
@sneridagh This is an intentional change, to avoid leaving empty text blocks lying around. Let me know if you want to discuss further. The bug with being able to add blocks with the Enter key even if that was disabled in the layout editor appears to be unrelated to this PR's changes.